docs/tasks/extra-impersonation-headers.md

```diff
@@ -0,0 +1,31 @@
+# Extra Impersonation Headers
+
+kube-oidc-proxy has support for adding 'extra' headers to the impersonation user
+info. This can be useful for passing extra information onto the target server
+about the proxy or client. kube-oidc-proxy currently supports two configuration
+options.
+
+# Client IP
+
+The following flag can be passed which will append the remote client IP as an
+extra header:
+
+`--extra-user-header-client-ip`
+
+Proxied requests will then contain the header
+`Impersonate-Extra-Remote-Client-Ip: <REMOTE_ADDR>` where  `<REMOTE_ADDR>` is
+the address of the client that made the request.
```

Comment:
I think we need to take care that this might not be the IP address of the actual end user.

We should maybe use https://github.com/sebest/xff/blob/master/xff.go to determine the IP address and/or proxy protocol support in the future and at least warn here that this is only the IP address which our connection is coming from